pkg/controller/container-runtime-config/helpers.go

```diff
@@ -753,3 +781,159 @@ func convertICSPToIDMS(icsp *apioperatorsv1alpha1.ImageContentSourcePolicy) *api
 		},
 	}
 }
+
+func ownerReferenceForImageConfig(imageConfig *apicfgv1.Image) metav1.OwnerReference {
+	return metav1.OwnerReference{
+		APIVersion: apicfgv1.SchemeGroupVersion.String(),
+		Kind:       "Image",
+		Name:       imageConfig.Name,
+		UID:        imageConfig.UID,
+	}
+}
+
+func ownerReferenceForImagePolicy(clusterImagePolicies []*apicfgv1alpha1.ClusterImagePolicy) metav1.OwnerReference {
+	if len(clusterImagePolicies) > 0 {
+		return metav1.OwnerReference{
+			APIVersion: apicfgv1alpha1.SchemeGroupVersion.String(),
+			Kind:       "ClusterImagePolicy",
+			Name:       clusterImagePolicies[0].Name,
+			UID:        clusterImagePolicies[0].UID,
+		}
+	}
+	return metav1.OwnerReference{}
+}
+
+func policyItemFromSpec(policy apicfgv1alpha1.Policy) (signature.PolicyRequirement, error) {
+	var (
+		sigstorePolicyRequirement signature.PolicyRequirement
+		signedIdentity            signature.PolicyReferenceMatch
+		signedOptions             []signature.PRSigstoreSignedOption
+		err                       error
+	)
+	switch policy.RootOfTrust.PolicyType {
+	case apicfgv1alpha1.PublicKeyRootOfTrust:
+		keyDataDec, err := b64.StdEncoding.DecodeString(policy.RootOfTrust.PublicKey.KeyData)
```

Comment:
The explicit base64 handling surprises me. Wouldn’t it be possible to make the API type a `[]byte`? At least https://github.com/kubernetes/api/blob/f3648a53522eb60ea75d70d36a50c799f7e4e23b/core/v1/types.go#L6838 suggests that might work.

Reply:
addressed https://github.com/openshift/api/pull/1765